ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts

```diff
@@ -287,71 +311,82 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
     );
 
     const verifyInputComponentSparseAddress$ = this.inputComponentAddress$.pipe(
-      filter(address => !this.enoughDataToVerify(address)),
-      tap(address => addressSuggestion$.next(null)),
-      map(address => [address, this.computeValidityForSparseAddress(address)]),
-      tap(([address, isValid]) => {
-        if (address && !isValid) {
-          this.stateUpdates$.next({ formErrorMessages: ['Invalid address'] });
-        } else {
-          this.stateUpdates$.next({ formErrorMessages: [] });
-        }
-      })
+        filter(address => !this.enoughDataToVerify(address)),
+        tap(address => addressSuggestion$.next(null)),
+        map(address => [address, this.computeValidityForSparseAddress(address)]),
+        tap(([address, isValid]) => {
+          if (address && !isValid) {
+            this.stateUpdates$.next({ formErrorMessages: ['Invalid address'] });
+          } else {
+            this.stateUpdates$.next({ formErrorMessages: [] });
+          }
+        })
     );
 
+    // Since we use switchMap, we can't expect every incoming value
+    // generates a corresponding one on other side
+    // so keep a count going in and subtract the total when something goes out
+    let initiatedVerifyAddressCalls = 0;
     const verifyInputComponentAddressAction$ = this.inputComponentAddress$.pipe(
-      filter(address => this.enoughDataToVerify(address)),
-      tap(address => addressSuggestion$.next(null)),
-      tap(() => busyCounter$.next(1)),
-      switchMap(inputAddress =>
-        this.addressService.verifyAddress(inputAddress).pipe(
-          map(verifyResponse => ({ entered: inputAddress,
-                                    suggested: new Address(verifyResponse),
-                                    warnings: verifyResponse.warnings }) as AddressSuggestion),
-          tap((addressSuggestion) => addressSuggestion$.next(addressSuggestion)),
-          catchError((error) => {
-            verificationError$.next(error);
-            return of(null);
-          })
-        )
-      ),
-      tap(() => busyCounter$.next(-1)),
+        filter(address => this.enoughDataToVerify(address)),
+        tap(address => addressSuggestion$.next(null)),
+        tap(() => busyCounter$.next(1)),
+        tap(() => ++initiatedVerifyAddressCalls),
+        switchMap(inputAddress =>
+            this.addressService.verifyAddress(inputAddress).pipe(
+                map(verifyResponse => ({
+                  entered: inputAddress,
+                  suggested: new Address(verifyResponse),
+                  warnings: verifyResponse.warnings
+                }) as AddressSuggestion),
+                tap((addressSuggestion) => addressSuggestion$.next(addressSuggestion)),
+                catchError((error) => {
+                  verificationError$.next(error);
+                  return of(null);
+                })
+            )
+        ),
+        tap(() => busyCounter$.next(-1 * initiatedVerifyAddressCalls))
     );
```

Comment:
Discovered that if we change values in fields while there is an ongoing /address/verify http going on, we would freeze the submit button forever. switchMap interrupts the inflight request and does not proceed.
Since for the busyCounter we are assuming that for every -1 there was a corresponing -1 , we would remain busy forever. This is the workaround to that situation.